tests/interfaces/conftest.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+from unittest.mock import patch
+
+import pytest
+from interface_tester import InterfaceTester
+from ops.pebble import Layer
+from scenario.state import Container, State
+
+from charm import TraefikIngressCharm
+
+
+@pytest.fixture
+def interface_tester(interface_tester: InterfaceTester):
+    with patch("charm.KubernetesServicePatch", lambda **unused: None):
```

Comment:
I wonder if it would be possible to pass a list of mocks that would apply throughout the tests, for example:
```python
interface_tester.run(
    mocks=[
        patch.object(MyCharm, "_version", property(lambda *_: "0.0.0")),
        patch.object(MyCharm, "_get_pvc_capacity"),
        ...
    ]
)
```

Reply:
what would the advantages of this design be in your opinion?

Reply:
@sed-i closing this as not relevant (it feels like a stylistic preference/implementation detail). If you want to discuss this further I'm open to that, but let's do that in charm-relation-interfaces or interface-tester-pytest.